cmd/rpcdaemon/commands/web3_api.go

```diff
@@ -0,0 +1,57 @@
+package commands
+
+import (
+	"github.com/ledgerwatch/turbo-geth/common"
+	"github.com/ledgerwatch/turbo-geth/common/hexutil"
+	"github.com/ledgerwatch/turbo-geth/crypto"
+	"github.com/ledgerwatch/turbo-geth/params"
+)
+
+// Web3API provides interfaces for the web3_ RPC commands
+type Web3API interface {
+	ClientVersion() (string, error)
+	Sha3(input hexutil.Bytes) hexutil.Bytes
+}
+
+type Web3APIImpl struct {
+}
+
+// NewWeb3APIImpl returns Web3APIImpl instance
+func NewWeb3APIImpl() *Web3APIImpl {
+	return &Web3APIImpl{}
+}
+
+// ClientVersion returns the node name
+func (api *Web3APIImpl) ClientVersion() (string, error) {
+	// https://infura.io/docs/ethereum/json-rpc/web3-clientVersion
+	return common.MakeName("TurboGeth", fullVersionStr()), nil
+}
+
+// New function so it always returns commit hash and date even if isRelease() is true
+func fullVersionStr() string {
+	vsn := params.VersionWithMeta
+	if len(gitCommit) > 0 {
+		vsn += "-" + gitCommit[:8]
+	}
+	if len(gitDate) > 0 {
```

Comment:
Let's not add date here, because I think reproducible builds are more important.

Reply:
I removed the date and now use `VersionWithCommit` with empty date string.